packages/mobile/src/screens/search-screen-v2/RecentSearches.tsx

```diff
@@ -1,20 +1,241 @@
+import { useDispatch, useSelector } from 'react-redux'
+
+import {
+  Button,
+  Flex,
+  IconButton,
+  IconClose,
+  Text,
+  spacing,
+  useTheme
+} from '@audius/harmony-native'
+import { FlatList, ProfilePicture } from 'app/components/core'
+import { CollectionImageV2 } from 'app/components/image/CollectionImageV2'
+import { TrackImageV2 } from 'app/components/image/TrackImageV2'
+import Skeleton from 'app/components/skeleton'
+import { UserLink } from 'app/components/user-link'
 
-import { Flex, Text } from '@audius/harmony-native'
+import type { AppTabScreenParamList } from '../app-screen'
 
 const { getV2SearchHistory } = searchSelectors
+const { removeItem, clearHistory } = searchActions
+
+type RecentSearchProps = {
+  children: React.ReactNode
+  to: To<AppTabScreenParamList>
+  searchItem: SearchItem
+}
+
+const RecentSearch = (props: RecentSearchProps) => {
+  const { children, to, searchItem } = props
+  const dispatch = useDispatch()
+
+  const handleClickRemove = useCallback(() => {
+    dispatch(removeItem({ searchItem }))
+  }, [dispatch, searchItem])
+
+  return (
+    <Link to={to}>
+      <Flex
+        direction='row'
+        w='100%'
+        justifyContent='space-between'
+        alignItems='center'
+        pv='s'
+      >
+        <Flex direction='row' gap='m'>
+          {children}
+        </Flex>
+        <IconButton
+          aria-label={messages.remove}
+          icon={IconClose}
+          color='subdued'
+          size='s'
+          onPress={handleClickRemove}
+        />
+      </Flex>
+    </Link>
+  )
+}
+
+const RecentSearchSkeleton = () => (
+  <Flex direction='row' w='100%' pv='s' justifyContent='space-between'>
+    <Flex direction='row' w='100%' gap='m'>
+      <Skeleton width={40} height={40} />
+
+      <Flex direction='column' gap='s'>
+        <Skeleton width={120} height={12} />
+        <Skeleton width={100} height={12} />
+      </Flex>
+    </Flex>
+  </Flex>
+)
+
+const RecentSearchTrack = (props: { searchItem: SearchItem }) => {
+  const { searchItem } = props
+  const { id } = searchItem
+  const { data: track, status } = useGetTrackById({ id })
+  const { data: user } = useGetUserById({ id: track?.owner_id ?? 0 })
+  const { spacing } = useTheme()
+
+  if (status === Status.LOADING) return <RecentSearchSkeleton />
+
+  if (!track) return null
+  const { permalink, title } = track
+
+  if (!user) return null
+
+  return (
+    <RecentSearch to={permalink} searchItem={searchItem}>
+      <TrackImageV2
+        trackId={id}
+        size={SquareSizes.SIZE_150_BY_150}
+        style={{
+          height: spacing.unit10,
+          width: spacing.unit10
+        }}
+      />
+      <Flex direction='column' alignItems='flex-start'>
+        <Text variant='body' size='s'>
+          {title}
+        </Text>
+        <Flex direction='row' alignItems='baseline'>
+          <Text variant='body' size='xs' color='subdued'>
+            {messages.track}
+            {' | '}
```

Comment:
i feel like this should be a divider, but this is good too

Reply:
oh yeah great idea, ill look into